File: app/src/main/java/com/example/data/CBSEQuestion.kt
package com.example.data

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "cbse_questions")
data class CBSEQuestion(
    @PrimaryKey(autoGenerate = true) val id: Int = 0,
    val clazz: String,         // e.g. "Class 10", "Class 12"
    val language: String,      // "English" or "Hindi"
    val subject: String,       // "Science", "Mathematics", "Physics", etc.
    val chapter: String,       // "Light - Reflection & Refraction" etc.
    val question: String,
    val option1: String,
    val option2: String,
    val option3: String,
    val option4: String,
    val correctIndex: Int,     // 0 for option1, 1 for option2, etc.
    val explanation: String
) {
    fun getOptions(): List<String> = listOf(option1, option2, option3, option4)
}

File: lib/models/cbse_question.dart
class CBSEQuestion {
  final int id;
  final String clazz;
  final String language;
  final String subject;
  final String chapter;
  final String question;
  final String option1;
  final String option2;
  final String option3;
  final String option4;
  final int correctIndex;
  final String explanation;

  CBSEQuestion({
    required this.id,
    required this.clazz,
    required this.language,
    required this.subject,
    required this.chapter,
    required this.question,
    required this.option1,
    required this.option2,
    required this.option3,
    required this.option4,
    required this.correctIndex,
    required this.explanation,
  });

  List<String> getOptions() => [option1, option2, option3, option4];

  Map<String, dynamic> toMap() {
    return {
      'clazz': clazz,
      'language': language,
      'subject': subject,
      'chapter': chapter,
      'question': question,
      'option1': option1,
      'option2': option2,
      'option3': option3,
      'option4': option4,
      'correctIndex': correctIndex,
      'explanation': explanation,
    };
  }

  factory CBSEQuestion.fromMap(Map<String, dynamic> map) {
    return CBSEQuestion(
      id: map['id'] ?? 0,
      clazz: map['clazz'] ?? "",
      language: map['language'] ?? "",
      subject: map['subject'] ?? "",
      chapter: map['chapter'] ?? "",
      question: map['question'] ?? "",
      option1: map['option1'] ?? "",
      option2: map['option2'] ?? "",
      option3: map['option3'] ?? "",
      option4: map['option4'] ?? "",
      correctIndex: map['correctIndex'] ?? 0,
      explanation: map['explanation'] ?? "",
    );
  }
}
